### PR TITLE
Histogram options

### LINKED ISSUE #599: 2d histogram options
We should review some of the stuff we added for 1-d in 2-d...

image

I'm not sure the "Sum X" is correct here? More like "Sum Z"

### PR DESCRIPTION
**Pull request by dmt0**
closes #599
- update packages

### CODE REVIEW

File: src/default_panels/GraphCreatePanel.js [hunk 1]
```diff
@@ -59,11 +63,26 @@ const GraphCreatePanel = (props, {localize: _}) => {
         options={[{label: _('Vertical'), value: 'v'}, {label: _('Horizontal'), value: 'h'}]}
       />
       <HistogramInfoVertical>
-        {_('Note: in vertical orientation, X values are used for bins and Y values for weights.')}
+        {_(
+          'Note: in vertical orientation, X values are used for bins and Y values for weights. You can configure the binning function in the '
```
Comment (nicolaskruchten) [inline src/default_panels/GraphCreatePanel.js]: wordsmithing ... :) "... X values are used for binning. If Y values are provided, they are used as inputs to the histogram function which you can configure in the Traces panel. If Y values are omitted, the histogram function defaults to Count."

Also please rename "binning function" to "histogram function" in the Traces panel.

Comment (nicolaskruchten) [reply src/default_panels/GraphCreatePanel.js]: Also: please move the "histogram function" outside of the "binning" section, right below normalization.

Comment (dmt0) [reply src/default_panels/GraphCreatePanel.js]: Done

**Comment by nicolaskruchten** [comment]
can you move the histogram function too plz ?

when that's done, 💃

**Comment by nicolaskruchten** [comment]
Hmmm I think the updated packages are a step too far here... I have to use `node` 10 now? Does that work on streambed?

Note also that even with node 10 I can't get `yarn install` to work.

```
nk$ rm -rf node_modules                                                                     ~p/react-chart-editor [mapbox-fixes]
nk$ yarn                                                                                    ~p/react-chart-editor [mapbox-fixes]
yarn install v1.9.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error eslint@5.4.0: The engine "node" is incompatible with this module. Expected version "^6.14.0 || ^8.10.0 || >=9.10.0".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
nk$ nvm use 9                                                                               ~p/react-chart-editor [mapbox-fixes]
Now using node v9.11.1 (npm v5.6.0)
nk$ yarn                                                                                    ~p/react-chart-editor [mapbox-fixes]
yarn install v1.9.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error postcss-combine-duplicated-selectors@6.0.2: The engine "node" is incompatible with this module. Expected version "^6.0.0 || ^8.0.0 || >=10.0.0".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
nk$ nvm use 10                                                                              ~p/react-chart-editor [mapbox-fixes]
Now using node v10.6.0 (npm v6.1.0)
nk$ yarn                                                                                    ~p/react-chart-editor [mapbox-fixes]
yarn install v1.9.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[-----------------------------------------------------------------------------------------------------------------------] 0/1695(node:91411) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[3/4] 🔗  Linking dependencies...
warning " > @plotly/draft-js-export-html@1.2.0" has unmet peer dependency "immutable@3.x.x".
warning " > draft-js-utils@1.2.0" has unmet peer dependency "immutable@3.x.x".
warning " > draft-js-import-html@1.2.1" has unmet peer dependency "immutable@3.x.x".
warning "draft-js-import-html > draft-js-import-element@1.2.1" has unmet peer dependency "immutable@3.x.x".
warning " > react-rangeslider@2.2.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0".
warning "@percy/react > @percy/react-percy-ci > @percy/react-percy-snapshot-render@0.4.1" has incorrect peer dependency "react@^15.4.0".
warning "@percy/react > @percy/react-percy-ci > @percy/react-percy-snapshot-render@0.4.1" has incorrect peer dependency "react-dom@^15.4.0".
warning " > css-loader@1.0.0" has incorrect peer dependency "webpack@^4.0.0".
[4/4] 📃  Building fresh packages...
[-/5] ⢀ waiting...
[-/5] ⢀ waiting...
[3/5] ⢀ fsevents
[4/5] ⢀ gl
error /Users/nicolas/plotly/react-chart-editor/node_modules/gl: Command failed.
Exit code: 1
Command: prebuild-install || node-gyp rebuild
Arguments:
Directory: /Users/nicolas/plotly/react-chart-editor/node_modules/gl
Output:
prebuild-install WARN install No prebuilt binaries found (target=10.6.0 runtime=node arch=x64 platform=darwin)
```

**Comment by nicolaskruchten** [comment]
OK I got it to run with a more recent 8.x

**Pull request merged**